internal/analytics/segment_reporter.go

```diff
@@ -59,10 +62,15 @@ func (r *SegmentReporter) RegisterCurrentIdentity(ctx context.Context, k8sCli ku
 
 // ReportCommand reports a new executed command. The command should be anonymized before using this method.
 // The RegisterCurrentIdentity needs to be called first.
-func (r *SegmentReporter) ReportCommand(platform config.CommPlatformIntegration, command string) error {
+func (r *SegmentReporter) ReportCommand(platform config.CommPlatformIntegration, command string, isInteractiveOrigin bool) error {
+	origin := typedCommand
+	if isInteractiveOrigin {
+		origin = buttonCommand
+	}
```

Comment:
This part of code tells me that we need an enum passed here instead of bool to have it extensible. I think we cannot assume interactive = button click. We can keep bool, but only if we want to have `interactive=true/false`. Will we have other triggers, like `selectOnChange` etc.? I can imagine such events 🤔


WDYT?

Reply:
I thought about it but at the same time it's not a problem to do that later as it's not user-facing API. Using bool right now simplifies a lot, see the code of slack in another places. The zero value of bool works here as intended. I would apply that once we will have such situation. However, the event that is sent to the segment.io has more generic properly, so we won't break this contract externally. So that's the reason I don't want to use the `interactive` properly externally, but "internally" I wanted to make it simple.

Reply:
👍 Alright, makes sense. I thought to do the internal work now, but yeah, let's postpone it until it's necessary 👍

Reply:
BTW consider changing `isInteractiveOrigin` to `isButtonClickOrigin` or something like this 🤔

Reply:
changed to `isButtonClickOrigin` 👍